File: Java/1119. Remove Vowels from a String.java
/**
 * https://leetcode.com/problems/remove-vowels-from-a-string/
 */

public class RemoveVowelsFromAString {
    // O(n) time complexity (n=s.length)
    // O(n) space complexity (n=s.length)
    String removeVowels(String s) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') {
                continue;
            }
            sb.append(c);
        }
        return sb.toString();
    }
}

class M {
    public static void main(String[] args) {
        RemoveVowelsFromAString r = new RemoveVowelsFromAString();
        String input = "leetcodeisacommunityforcoders";
        System.out.println(r.removeVowels(input)); // output: ltcdscmmntyfrcdrs
    }
}

File: Kotlin/1119. Remove Vowels from a String.kt
/**
 * https://leetcode.com/problems/remove-vowels-from-a-string/
 */

// O(n) time complexity (n=s.length)
// O(n) space complexity (n=s.length)
fun removeVowels(s: String): String {
    val sb = StringBuilder()
    s.forEach { c ->
        if (c != 'a' && c != 'e' && c != 'i' && c != 'o' && c != 'u') {
            sb.append(c)
        }
    }
    return sb.toString()
}

fun main() {
    var input = "aeiou"
    println(removeVowels(input)) // output:

    input = "leetcodeisacommunityforcoders"
    println(removeVowels(input)) // output: ltcdscmmntyfrcdrs
}